.github/workflows/npm-publish-beta.yml

```diff
@@ -0,0 +1,54 @@
+name: Build and Publish NPM Package
+
+on:
+    push:
+        branches:
+            - develop
+
+jobs:
+    deploy-beta:
+        runs-on: ubuntu-latest
+        if: github.ref == 'refs/heads/develop'
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v3
+
+            - name: Set up Node.js
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 16
+                  registry-url: https://registry.npmjs.org/
+
+            - name: Install dependencies
+              run: npm install
+
+            - name: Build
+              run: npm ci && npm run build:prod
+
+            - name: Publish to npm (beta version)
+              run: npm run npm:publish:beta
+              env:
+                  NODE_AUTH_TOKEN: ${{secrets.NPM_AUTH_TOKEN}}
+    deploy-release:
+        runs-on: ubuntu-latest
+        if: github.ref == 'refs/heads/master'
```

Comment:
Also, the default branch is `main` but you used `master`, would that change anything?